strutil/shlex/shlex.go

```diff
@@ -414,3 +415,28 @@ func Split(s string) ([]string, error) {
 		subStrings = append(subStrings, word)
 	}
 }
+
+// Join returns a shell-escaped string version of the slice s
+func Join(s []string) string {
+	var quoted []string
+	for _, subString := range s {
+		quoted = append(quoted, Quote(subString))
+	}
+	return strings.Join(quoted, " ")
+}
+
+var unsafeQuoteChars = regexp.MustCompile(`[^\w@%+=:,./-]`)
```

Comment:
maybe you could transform `unsafeQuoteChars` to a `const`?

Reply:
> maybe you could transform `unsafeQuoteChars` to a `const`?

IIRC `regexp.MustCompile` will return `*regexp.Regexp` which cannot be a constant